src/model/page-config-resolver.js

```diff
@@ -283,13 +283,180 @@ class JsonLdParser {
   }
 }
 
+/**
+ * Class the describes the microdata found in the page
+ */
+class Microdata {
+  /**
+   * @param {!Array<!{id: number, properties: Object<string, (string|boolean)>}>} entries
+   */
+  constructor(entries) {
+    this.entries = entries;
+  }
+
+  /**
+   * Returns the first page configuration found, if present
+  * @return {?PageConfig} pageConfig found
+  */
+  getPageConfig() {
+    const pageConfigs = [];
+    let locked = false;
+    let productId = null;
+    this.entries.forEach(item => {
+      Object.keys(item.properties).forEach(property => {
+        if (property == 'isAccessibleForFree') {
+          locked = !item.properties[property];
+        }
+        if (property == 'productID') {
+          productId = item.properties[property];
+        }
+      });
+      if (productId != null) {
+        pageConfigs.push(new PageConfig(productId, locked));
+      }
+    });
+    if (pageConfigs.length == 0) {
+      return null;
+    } else {
+      return pageConfigs[0];
+    }
+  }
+}
+
 class MicrodataParser {
   /**
    * @param {!Doc} doc
    */
   constructor(doc) {
     /** @private @const {!Doc} */
     this.doc_ = doc;
+    /** @private {?boolean} */
+    this.access_ = null;
+    /** @private {?string} */
+    this.productID_ = null;
+  }
+
+  /**
+   * Returns false if access is restricted, otherwise true
+   * @param {Node} root A node that is an item of type 'NewsArticle'
+   * @return {?boolean} locked access
+   * @private
+   */
+  discoverAccess_(root) {
+    const ALREADY_SEEN = 'alreadySeenForAccessInfo';
+    const nodeList = root
+        .querySelectorAll("[itemprop='isAccessibleForFree']");
+    for (let i = 0; nodeList[i]; i++) {
+      const element = nodeList[i];
+      const content = element.content || element.textContent;
+      if (!content) {
+        continue;
+      }
+      let accessForFree = null;
+      if (content.toLowerCase() == 'true') {
+        accessForFree = true;
+      } else if (content.toLowerCase() == 'false') {
+        accessForFree = false;
+      }
+      if (this.isValidElement_(element, root, ALREADY_SEEN)) {
+        return accessForFree;
+      }
+    }
+    return null;
+  }
+
+  /**
+   * Verifies if a node is valid based on the following
+   * - child of an item of type 'NewsArticle'
+   * - not a child of an item of type 'Section'
+   * - not seen before, marked using the alreadySeen tag
+   * @param current the node to be verified
+   * @param root the parent to track up to
+   * @param alreadySeen used to tag already visited nodes
+   * @return {!boolean} valid node
+   * @private
+   */
+  isValidElement_(current, root, alreadySeen) {
+    for (let node = current;
+        node && !node[alreadySeen]; node = node.parentNode) {
+      node[alreadySeen] = true;
+      if (node.hasAttribute('itemscope')) {
+        const type = node.getAttribute('itemtype');
```

Comment:
It could be null. Do `const type = node.getAttribute(...) || ''`. Or change `if` above to include `if (node.hasAttribute('itemscope') || node.hasAttribute('itemtype'))`